gson/src/main/java/com/google/gson/stream/JsonWriter.java

```diff
@@ -162,6 +167,190 @@
     HTML_SAFE_REPLACEMENT_CHARS['\''] = "\\u0027";
   }
 
+  private class StringValWriter extends Writer {
+    private boolean isClosed;
+
+    private void verifyNotClosed() throws IOException {
+      if (isClosed) {
+        throw new IOException("Writer is closed");
+      }
+    }
+
+    @Override
+    public void write(int c) throws IOException {
+      verifyNotClosed();
+      stringPiece((char) c);
+    }
+
+    /**
+     * @param len
+     *      Length of the data, e.g. array length
+     * @param off
+     *      0-based offset where the section begins
+     * @param sectLen
+     *      Length of the section
+     * @throws IndexOutOfBoundsException
+     *      If {@code off} or {@code sectLen} is invalid
+     */
+    private void validateIndices(int len, int off, int sectLen) {
```

Comment:
len -> length
off --> offset
sectLen --> sectionLength

Reply:
The `write` methods use abbreviated names as well since the `java.io.Writer` methods do this. Do you want all parameter names written out? In my opinion especially here this does not really improve readability because the method name indicates their meaning.

Reply:
yes, these are the rules google follows.

Reply:
Alright, I have written out the names of fields and parameters I introduced. However, for overridden `Reader`  and `Writer` methods I am still using the JDK names. Guava seems to be doing the same, see for example [AppendableWriter](https://github.com/google/guava/blob/master/guava/src/com/google/common/io/AppendableWriter.java).

Please let me know if you have other change requests.

Reply:
Alright, I have written out the names of fields and parameters I introduced. However, for overridden `Reader`  and `Writer` methods I am still using the JDK names. Guava seems to be doing the same, see for example [AppendableWriter](https://github.com/google/guava/blob/master/guava/src/com/google/common/io/AppendableWriter.java).

Please let me know if you have other change requests.